src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -381,6 +389,60 @@ public string Add(DiagnosticDescriptor descriptor)
                 list.Sort((x, y) => string.CompareOrdinal(x.Key, y.Key));
                 return list;
             }
+
+            /// <summary>
+            /// Compares descriptors by the values that we write to the log and nothing else.
+            /// 
+            /// We cannot use the IEquatable implementation because it includes message format 
+            /// and we don't write it out. It also ignores tags, which we do write.
+            /// </summary>
```

Comment:
Why do you say it ignores tags? Are "tags" different from the "CustomTags" which you do compare below?